libopenage/pathfinding/grid.cpp

```diff
@@ -64,6 +64,39 @@ const std::vector<std::shared_ptr<Sector>> &Grid::get_sectors() const {
 	return this->sectors;
 }
 
+
+const Grid::portalgraph_t &Grid::get_portal_graph() const {
+	return this->portal_graph;
+}
+
+void Grid::init_portal_graph()
+{
+	//map
+	for(auto& sector: this->get_sectors())
+	{
+		for(auto& portal : sector->get_portals())
+		{
+			//node_t portal_node = std::make_shared<PortalNode>(portal, sector->get_id(), nullptr);
+			std::map<portal_id_t, int> links = this->portal_graph[portal->get_id()];
```

Comment:
Is there a reason why you used  `map` and not `unordered_map` here? `map` has linear lookup time, while `unordered_map`'s lookup time is constant, so the latter would be faster.

Reply:
Hey, my thinking was that an ordered map would provide the closest neighbor first and this might provide some performance benefit. This would have to be tested ofcourse.

Reply:
Generally, I would recommend to use the most viable default type first (e.g. unordered map) and later measure alternatives for speed. Otherwise you might end up with a lot of premature optimization which could make the code slower even.

A vector of these items, sorted by closest neighbor, might also work a little better than a map for speed, since it's memory is contiguos. But let's try that later.